src/menu.rs

```diff
@@ -6,37 +6,145 @@ use std::{
   hash::{Hash, Hasher},
 };
 
-#[derive(Debug, Clone)]
-pub struct Menu {
-  pub title: String,
-  pub items: Vec<MenuItem>,
+use crate::platform_impl::{CustomMenuItem as CustomMenuItemPlatform, Menu as MenuPlatform};
+
+/// Object that allows you to build a `ContextMenu`, for the tray.
+pub struct ContextMenu(pub(crate) Menu);
+/// Object that allows you to build a `MenuBar`, for the *Window* menu in Windows and Linux and the *Menu bar* on macOS.
+pub struct MenuBar(pub(crate) Menu);
+
+/// A custom menu item.
+pub struct CustomMenuItem<'a> {
+  id: MenuId,
+  title: &'a str,
+  keyboard_accelerator: Option<&'a str>,
+  enabled: bool,
+  selected: bool,
```

Comment:
This field could be `Option<bool>` to represent if the item should be a select item or not, and then select it or not with `Some(true)` and `Some(false)`.

Reply:
i think it’s fair, im not against

Reply:
What's about `MenuItemBuilder`

```rust
tray_menu.add_item(MenuItem::new("Focus window").with_selected(true));
```

rename `MenuItem` for `NativeItem` as it's used by `fn add_native_item`

Reply:
Sounds good